pkg/R/DataFrame.R

```diff
@@ -52,21 +53,108 @@ dataFrame <- function(sdf) {
 #' printSchema(df)
 #'}
 
-setGeneric("printSchema", function(df) { standardGeneric("printSchema") })
+setGeneric("printSchema", function(x) { standardGeneric("printSchema") })
 
 setMethod("printSchema",
-          signature(df = "DataFrame"),
-          function(df) {
-            sdf <- df@sdf
-            schemaString <- callJMethod(sdf, "printSchema")
+          signature(x = "DataFrame"),
+          function(x) {
+            schemaString <- callJMethod(x@sdf, "printSchema")
             cat(schemaString)
           })
 
+#' Get schema object
+#' 
+#' Returns the schema of this DataFrame as a list of named lists. Each named
```

Comment:
Sorry, I missed them in the begging. If we do it in this way, we will need to serialize all the DateType and re-construct all of them in R. We do it in this way in Python, but don't think it's a good idea in R, because OO is much harder in R than Python.

So, I'd like to have a wrapper for DataType in JVM, it will much easier.